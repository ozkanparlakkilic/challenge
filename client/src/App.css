.App {
    text-align: center;
    height: 100%;
    display: flex;
    flex-direction: column;
}

.d-flex {
    display: flex;
}

.justify-content-center {
    justify-content: center;
}

.justify-content-end {
    justify-content: end;
}

.justify-content-start {
    justify-content: start;
}

.align-items-center {
    align-items: center;
}

.flex-direction-column {
    flex-direction: column;
}

.justify-content-between {
    justify-content: space-between;
}

.flex-1 {
    flex: 1;
}

.h-100 {
    height: 100%;
}

.w-100 {
    width: 100%;
}

.align-self-center {
    align-self: center;
}

.align-items-start {
    align-items: start;
}

.align-items-end {
    align-items: end;
}
