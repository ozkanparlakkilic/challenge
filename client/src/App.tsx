import React from 'react';
import './App.css';
import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { Home, Login, QuizList, Register } from './pages';
import { Header } from './components/common';
import { MainLayout } from './layouts';
import { useAppSelector } from './store/hooks';

function App() {
    const user = useAppSelector(state => (state.user))

    return (
        <div className="App">
            <Router>
                <Header />
                <MainLayout>
                    {!user.data ? (
                        <Routes>
                            <Route path="/" element={<Login />} />
                            <Route path="login" element={<Login />} />
                            <Route path="register" element={<Register />} />
                        </Routes>
                    ) : (
                        <Routes>
                            <Route path="/" element={<Home />} />
                            <Route path="quiz-list" element={<QuizList />} />
                        </Routes>
                    )}
                </MainLayout>
            </Router>
        </div>
    );
}

export default App;
