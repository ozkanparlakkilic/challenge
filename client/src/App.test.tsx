import React from 'react';
import { render } from '@testing-library/react';
import App from './App';


describe('Renderings', () => {
    it('should render the child component', () => {
      const { container } = render(<App />)
      expect(container).toMatchSnapshot();
    });
});
