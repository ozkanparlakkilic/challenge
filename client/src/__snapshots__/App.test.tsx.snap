// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderings should render the child component 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <div
        class="d-flex justify-content-between align-items-center header_nav"
      >
        <div
          class="d-flex flex-1"
        >
          <svg
            class="icons"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </div>
        <div
          class="d-flex flex-1 justify-content-center"
        >
          <h3
            class="header_title"
          >
            Challenge Me
          </h3>
        </div>
        <div
          class="d-flex flex-1 align-items-center justify-content-end"
        >
          <button
            class=" btn"
          >
            Sign In
          </button>
          <button
            class=" btn"
          >
            Sign Up
          </button>
        </div>
      </div>
    </header>
    <div
      class="flex-1 main_layout"
    >
      <div
        class=" d-flex flex-direction-column justify-content-center align-items-center h-100 content_layout"
      >
        <div
          class=" auth_layout"
        >
          <div>
            <img
              alt="Challenge Me"
              height="170px"
              src="https://upload.wikimedia.org/wikipedia/commons/d/db/Challenge_Logo.png"
              style="margin: 40px 0px;"
              width="250px"
            />
          </div>
          <form
            autocomplete="off"
            class=" form"
          >
            <div
              class="w-100 d-flex flex-direction-column align-items-start"
            >
              <input
                autocomplete="off"
                class=" input"
                placeholder="Kullanıcı adı giriniz"
                type="text"
              />
            </div>
            <div
              class="w-100 d-flex flex-direction-column align-items-start"
            >
              <input
                autocomplete="off"
                class=" input"
                placeholder="Password giriniz"
                type="password"
              />
            </div>
            <button
              class="align-self-center btn"
              style="width: 50%; margin: 20px;"
            >
              Sign Up
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
